build-scripts/for-osx/build-on-osx.sh

```diff
@@ -23,10 +22,23 @@ pushd build/osx
 rm -rf *
 export LANG=C
 
-OS_PRMS="-DDOCBOOK_DIR=/usr/local/opt/docbook-xsl/docbook-xsl -DCMAKE_OSX_DEPLOYMENT_TARGET=12.1"
+case `arch` in
+  arm64) # Apple silicon
+  OS_PRMS="-DDOCBOOK_DIR=/opt/homebrew/opt/docbook-xsl/docbook-xsl"
+  ;;
+  i386) # Intel
+  OS_PRMS="-DDOCBOOK_DIR=/usr/local/opt/docbook-xsl/docbook-xsl -DCMAKE_OSX_DEPLOYMENT_TARGET=12.1"
+  ;;
+esac
 GO_PRMS="-DCMAKE_BUILD_TYPE=Release $CMAKE_VERSION_PRMS"
 cmake -G "Unix Makefiles" $OS_PRMS $GO_PRMS . $SRC_DIR
 make -k $PARALLEL_PRMS VERBOSE=1 package
 
-popd
+codesign --force --sign - GrandOrgue.app/Contents/Frameworks/*.*
```

Comment:
The signing steps are present twice: here and in the cpack scripts.

Could you create a single cmake script that would make signing and would be called from cpack and as a custom build target from make? See https://cmake.org/cmake/help/latest/command/add_custom_target.html and https://dev.to/iblancasa/learning-cmake-3-understanding-addcustomcommand-and-addcustomtarget-43gp

Reply:
How do I add a custom build target that is always built and doesn't start building until all other build targets and other build steps are finished?

Reply:
```
add_custom_target(
  MacOsApplication    # the name of the custom target
  ALL
  DEPENDS GrandOrgue GrandOrgueTool GrandOrguePerfTest resources # run after building these targets
  COMMAND "${CMAKE_COMMAND} " -P "${CMAKE_SOURCE_DIR}/SignMacOsApp.cmake" "-DAPP_DIR="${PROJECT_NAME}.app" # add other parameters if it needs
)
```